akka-docs/src/main/paradox/typed/cluster-sharding.md

```diff
@@ -125,10 +220,128 @@ message if the entity needs to perform some asynchronous cleanup or interactions
+
+To make the list of entities in each `Shard` persistent (durable):
+
+1. set the `rememberEntities` flag to true in `ClusterShardingSettings` when 
+starting a shard region (or its proxy) for a given `entity` type,
+or configure `akka.cluster.sharding.remember-entities = on`, 
+1. make sure the it is possible to extract the ID of the sharded entity in the message.
+
+The performance cost of `rememberEntities` is rather high when starting/stopping entities and when
+shards are rebalanced. This cost increases with number of entities per shard, thus it is not
+recommended with more than 10000 active entities per shard.  
+
+### Behavior When Enabled 
+
+When `rememberEntities` is enabled, whenever a `Shard` is rebalanced onto another
+node or recovers after a crash it will recreate all the entities which were previously
+running in that `Shard`. To permanently stop entities, a `Passivate` message must be
+sent to the parent of the entity actor, otherwise the entity will be automatically
+restarted after the entity restart backoff specified in the configuration.
+
+When [Distributed Data mode](#distributed-data-mode) is used the identifiers of the entities are
+stored in @ref:[Durable Storage](distributed-data.md#durable-storage) of Distributed Data. You may want to change the
+configuration of the `akka.cluster.sharding.distributed-data.durable.lmdb.dir`, since
+the default directory contains the remote port of the actor system. If using a dynamically
+assigned port (0) it will be different each time and the previously stored data will not
+be loaded.
+
+The reason for storing the identifiers of the active entities in durable storage, i.e. stored to
+disk, is that the same entities should be started also after a complete cluster restart. If this is not needed
+you can disable durable storage and benefit from better performance by using the following configuration:
+
+```
+akka.cluster.sharding.distributed-data.durable.keys = []
+```
+
+### Behavior When Not Enabled 
+
+When `rememberEntities` is disabled (the default), a `Shard` will not automatically restart any entities
+after a rebalance or recovering from a crash. Instead, entities are started once the first message
+for that entity has been received in the `Shard`.
+
+### Startup after minimum number of members
+
+It's recommended to use Cluster Sharding with the Cluster setting `akka.cluster.min-nr-of-members` or
+`akka.cluster.role.<role-name>.min-nr-of-members`. `min-nr-of-members` will defer the allocation of the shards
+until at least that number of regions have been started and registered to the coordinator. This
+avoids that many shards are allocated to the first region that registers and only later are
+rebalanced to other nodes.
+
+See @ref:[How To Startup when Cluster Size Reached](cluster.md#how-to-startup-when-a-cluster-size-is-reached)
+for more information about `min-nr-of-members`.
+
+## Removal of internal Cluster Sharding data
```

Comment:
This is only relevant for "Persistent Mode" so start the section by mentioning that.